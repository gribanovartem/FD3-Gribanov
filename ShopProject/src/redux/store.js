import { createStore } from 'redux'
import combinedReducer from './reducers.js'

const store = createStore(combinedReducer)

export default store