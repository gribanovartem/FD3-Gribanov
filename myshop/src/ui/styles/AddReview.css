.mainDiv .addReview {
    position: fixed;
    top: 0;
    transition-duration: 0.5s;
}
.addReview form {
    margin: 5%;
}
.addReview .button {
    margin-right: 1%;
}
.modal {
    display: block;
    /* position: absolute; */
    height: auto;
    top: -2000px;
    transition-duration: 2.5s;
}