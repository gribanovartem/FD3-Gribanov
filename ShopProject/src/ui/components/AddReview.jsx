import React from 'react'
import '../styles/AddReview.css'
import { connect } from 'react-redux'
import PropTypes from "prop-types"
import * as firebase from "firebase/app"
import { set_reviews, set_mode_0, } from '../../redux/reviewsAC'
import 'firebase/storage' 

class AddReview extends React.Component {
  constructor(props) {
    super(props)
    this.inputName = React.createRef()
    this.inputCity = React.createRef()
    this.inputReview = React.createRef()
  }

    closeModal=() => {
      this.props.dispatch(set_mode_0())
    }

    submitReview=() => {
      const name = this.inputName.current.value
      const city = this.inputCity.current.value
      const review = this.inputReview.current.value
      const newReview = {
        "name": name,
        "city": city,
        "review": review,
      }
      const config = {
        apiKey: "AIzaSyD9NIAz1hpx5LdZTLMrhMga1l3rQkqYZHc",
        authDomain: "shop-gribanov.firebaseapp.com",
        databaseURL: "https://shop-gribanov.firebaseio.com",
        projectId: "shop-gribanov",
        storageBucket: "shop-gribanov.appspot.com",
        messagingSenderId: "161416264169",
        appId: "1:161416264169:web:a05312c302f35c59f1f378",
      }
      
      firebase.initializeApp(config)
      const newReviews = [newReview,...this.props.reviews.reviews]
      const storage = firebase.storage()
      console.log(storage)
      var storageRef = storage.ref()
      var reviewsRef = storageRef.child('reviews.json')
      const blob = new Blob([JSON.stringify(newReviews, null, 2)], { type : 'application/json' })
      reviewsRef.put(blob).then(() => {
        this.props.dispatch(set_reviews(newReviews))
        this.props.dispatch(set_mode_0())
      })
    }

    render() {
      return (
        <div className={this.props.reviews.mode===0?"modal":"modal addReview"} id="exampleModalCenter" tabIndex="-1" role="dialog" aria-labelledby="exampleModalCenterTitle" aria-hidden="true">
          <div className="modal-dialog modal-dialog-centered" role="document">
            <div className="modal-content">
              <div className="modal-header">
                <h5 className="modal-title" id="exampleModalLongTitle">Добавить отзыв</h5>
                <button type="button" className="close" data-dismiss="modal" aria-label="Close" onClick={this.closeModal}>
                  <span aria-hidden="true">&times;</span>
                </button>
              </div>
              <div className="modal-body">
                <form className="form">
                  <div className="form-group">
                    <label htmlFor="UserName">Имя</label>
                    <input className="form-control" type="text" id="UserName" ref={this.inputName} />
                  </div>
                  <div className="form-group">
                    <label htmlFor="City">Город</label>
                    <input className="form-control" type="text" id="City" ref={this.inputCity} />
                  </div>
                  <div className="form-group">
                    <label htmlFor="exampleFormControlTextarea1">Ваш отзыв</label>
                    <textarea className="form-control" id="exampleFormControlTextarea1" rows="3" ref={this.inputReview} />
                  </div>
                </form>
              </div>
              <div className="modal-footer">
                <button type="button" className="btn btn-secondary" data-dismiss="modal" onClick={this.closeModal}>Отмена</button>
                <button type="button" className="btn btn-primary" onClick={this.submitReview}>Отправить отзыв</button>
              </div>
            </div>
          </div>
        </div>
         
      )
    }
}
const mapStateToProps = function (state) {
  return {
    reviews: state.reviews,
  }
}
AddReview.propTypes = {
  reviews: PropTypes.shape({
    reviews: PropTypes.array,
    mode: PropTypes.number.isRequired,
  }),
  mode: PropTypes.func,
  dispatch: PropTypes.func.isRequired,
}
AddReview.defaultProps = {
  reviews: PropTypes.shape({
    reviews: PropTypes.number,
  }),
  mode: PropTypes.func,
}
export default connect(mapStateToProps)(AddReview)