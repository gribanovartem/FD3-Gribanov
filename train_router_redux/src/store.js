import { createStore } from 'redux';
import combinedReducer from './redux/reducers.js';

let store = createStore(combinedReducer);

export default store;