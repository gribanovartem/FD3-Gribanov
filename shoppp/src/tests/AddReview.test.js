import React from 'react';
import { act } from 'react-dom/test-utils';
import AddReview from '../ui/components/AddReview';
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import store from '../redux/store'
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';

// let container;
// beforeEach(() => {
//   container = document.createElement('div');
//   document.body.appendChild(container);
// });

// afterEach(() => {
//   document.body.removeChild(container);
//   container = null;
// });
configure({ adapter: new Adapter() });
describe('AddReview renders correctly', () => {
  const componentSnap = renderer.create(<Provider store={store}>
                                          <AddReview/>
                                        </Provider>);
  let componentTree=componentSnap.toJSON();
  expect(componentTree).toMatchSnapshot();
  it('should render MobileAddOrEdit component when clicked', () => {
    const component = renderer.create(<Provider store={store}>
                                <AddReview/>
                              </Provider>);
    const Link = component.root.findByProps({className: 'close'});
    Link.props.onClick();
    

    componentTree=componentSnap.toJSON();
    expect(componentTree).toMatchSnapshot();
    });

  // expect(container).toMatchSnapshot();
  // add1=container.toJSON();
  // const buttonClose = add1.root.findByProps({className: 'close'}); 
  // buttonClose.props.onClick();
  
  // expect(add).toMatchSnapshot();
});