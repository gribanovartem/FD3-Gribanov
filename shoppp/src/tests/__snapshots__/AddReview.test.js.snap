// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalCenterTitle"
  className="modal"
  id="exampleModalCenter"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog modal-dialog-centered"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLongTitle"
        >
          Добавить отзыв
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <form
          className="form"
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="UserName"
            >
              Имя
            </label>
            <input
              className="form-control"
              id="UserName"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="City"
            >
              Город
            </label>
            <input
              className="form-control"
              id="City"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="exampleFormControlTextarea1"
            >
              Ваш отзыв
            </label>
            <textarea
              className="form-control"
              id="exampleFormControlTextarea1"
              rows="3"
            />
          </div>
        </form>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Отмена
        </button>
        <button
          className="btn btn-primary"
          onClick={[Function]}
          type="button"
        >
          Отправить отзыв
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`AddReview renders correctly should render MobileAddOrEdit component when clicked 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalCenterTitle"
  className="modal"
  id="exampleModalCenter"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog modal-dialog-centered"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLongTitle"
        >
          Добавить отзыв
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <form
          className="form"
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="UserName"
            >
              Имя
            </label>
            <input
              className="form-control"
              id="UserName"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="City"
            >
              Город
            </label>
            <input
              className="form-control"
              id="City"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="exampleFormControlTextarea1"
            >
              Ваш отзыв
            </label>
            <textarea
              className="form-control"
              id="exampleFormControlTextarea1"
              rows="3"
            />
          </div>
        </form>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Отмена
        </button>
        <button
          className="btn btn-primary"
          onClick={[Function]}
          type="button"
        >
          Отправить отзыв
        </button>
      </div>
    </div>
  </div>
</div>
`;
